### PR TITLE
Event-based STOMP header generation

### LINKED ISSUE #1856: STOMP message headers are not alterable
Looking into adjusting STOMP headers for a couple of reasons (including making things persistent from another module, to address #1857, as well as making some bigger tweaks to authentication); however, the [STOMP headers are being set explicitly](https://github.com/Islandora/islandora/blob/cf82e8f392c081221781fd568cfc11d1bde7fd87/src/EventGenerator/EmitEvent.php#L151).

As such, looking to introduce some other mechanism such that they might be manipulated, via hooks or events. Events seems like the most flexible with priorities, and is the mechanism already in play to generate JWTs.

### PR DESCRIPTION
**Pull request by adam-vessey**
**JIRA Ticket**: N/A

* Other Relevant Links (Google Groups discussion, related pull requests, Release pull requests, etc.)

Fixes https://github.com/Islandora/documentation/issues/1856

# What does this Pull Request do?

Introduces an event-based mechanism to generate headers for STOMP messages, so other modules can subscribe to the event to influence what headers are set.

# What's new?

We dispatch a new `StompHeaderEvent` (`islandora.stomp.header_event`) event, which can be subscribed to in other modules in order to add/alter the headers which get used for the STOMP message.

There's a base subscriber for this new event included with a very low priority which maintains the current behaviour of generating and adding the token, but it has been made to avoid overwriting an `Authorization` token, if there's already one set.

One divergence is the addition of the `aud` claim; however, the implementation has allowed JWTs _without_ this claim to continue being accepted, expecting that we will remove this ability in the future.

May be a small PR in the future making use of this to address https://github.com/Islandora/documentation/issues/1857

# How should this be tested?

Pretty much just regression test, creating some content with derivatives, and ensure that:
* The derivatives are created
* Things are indexed as expected
* Things are deleted from the index accordingly.

# Additional Notes:
Any additional information that you think would be helpful when reviewing this PR.

Example:
* **Does this change the interface, add a new feature, or otherwise change behaviours that would require updating documentation?** No?
* **Does this change add any new dependencies?** No.
* **Does this change require any other modifications to be made to the repository (ie. Regeneration activity, etc.)?** No.
* **Could this change impact execution of existing code?** Unlikely.

# Interested parties
@Islandora/8-x-committers

### CODE REVIEW

File: src/Plugin/Action/EmitFileEvent.php [hunk 1]
```diff
@@ -2,17 +2,11 @@
...
 
+use Drupal\islandora\EventGenerator\EmitEvent;
+
 use Drupal\Core\Entity\EntityInterface;
-use Drupal\Core\Entity\EntityTypeManagerInterface;
 use Drupal\Core\File\FileSystemInterface;
-use Drupal\Core\Messenger\MessengerInterface;
-use Drupal\Core\Session\AccountInterface;
 use Drupal\Core\Site\Settings;
-use Drupal\Core\StreamWrapper\StreamWrapperManager;
```
Comment (whikloj) [inline src/Plugin/Action/EmitFileEvent.php:11]: `StreamWrapperManager` is still used on line 109/53, I tried to add an audio media and got a WSOD. 

Stacktrace in the apache log refers to  
```
... Error: Class 'Drupal\\islandora\\Plugin\\Action\\StreamWrapperManager' not found in /home/vagrant/all_claw/islandora/src/Plugin/Action/EmitFileEvent.php ...
```

**Review by whikloj — CHANGES_REQUESTED** [review]
This works fine for content nodes, but adding media is failing.

File: src/Plugin/Action/EmitFileEvent.php (hunk 1 above)
Comment (adam-vessey) [reply src/Plugin/Action/EmitFileEvent.php:11]: Ah... stuff was originally developed against (a fork of) 1.1.1... which was still using [the (now deprecated) `$fileSystem->uriScheme()` thing](https://api.drupal.org/api/drupal/core%21lib%21Drupal%21Core%21File%21FileSystemInterface.php/function/FileSystemInterface%3A%3AuriScheme/8.9.x)... patch applied relatively cleanly (IIRC), and the actions were giving a checkmark... I guess this class isn't actually covered under any (unit) tests?

Anyway... the `$fileSystem` being injected here is no longer being used. Should we drop it? Would rather simplify the class. [Removal should be inline with Drupal's policies WRT protected properties?](https://www.drupal.org/about/core/policies/core-change-policies/drupal-8-and-9-backwards-compatibility-and-internal-api#protected-properties):

> Protected properties on a class are always considered `@internal` unless they're on a base class marked with `@api`

Comment (whikloj) [reply src/Plugin/Action/EmitFileEvent.php:11]: Probably no test, if you want to add a unit test that would be 🥇 . As for the `$fileSystem`, you are correct that it does appear to not be necessary, I'd vote to get rid of it and simplify.

Comment (adam-vessey) [reply src/Plugin/Action/EmitFileEvent.php:11]: Rebased on top of `HEAD`, and added the reference here back in... left `$fileSystem` (for now?).

Comment (adam-vessey) [reply src/Plugin/Action/EmitFileEvent.php:11]: _Did_ take a look at how tests might be written for this `::generateData()` stuff; however, the method is `protected`... would get into either changing the visibility (and pushing such a change up into the interface and propagating around), using reflection to adjust the visibility when running the test (which is relatively simple but gross) or mocking/loading up all the services required to `::execute()` the event, which... is... rather more complex.

... start wondering if things should be structured differently, to facilitate testing?... anyway...

**Review by whikloj — APPROVED** [review]
👍

**Pull request merged**